dna/clutter/clutter.js

```diff
@@ -42,15 +42,172 @@ function post(post) {
     var me = getMe();                       // Looks up my hash address and assign it to 'me'
                                             // which DHT nodes will use to request validation info from my source chain
-
-      // On the DHT, puts a link on my hash to the new post
     commit("post_links",{Links:[{Base:me,Link:key,Tag:"post"}]});
 
+    // TODO detect a hash
+    debug(post);
+    debug(post.message);
+    debug("Starting HASHtag search");
+  //  var hashtag=[];
+    hashTag=detectHashtag(post.message);
+//  var a=hashtag[0];
+
+     if (hashTag != null)
+    {
+      debug(hashTag);
+      debug("::Hashtag found::");
+      //TODO searchHashTag(hashtag)}
+//  debug("HAHSTAG Length"+hashTag.length);
+    searchHashTag(hashTag,post);
+      debug("HASHTAGE SAVED");
+}
+ else {
+      debug("Hashtag not found");
+    }      // On the DHT, puts a link on my hash to the new post
+
+
     debug("meta: "+JSON.stringify(getLink(me,"post",{Load:true})));
     debug(key);
     return key;                                  // Returns the hash key of the new post to the calling function
 }
 
+function searchHashTag(hashTag,post)
+{
+ for (var i = 0; i < hashTag.length; i++)
+ {
+//   debug("SEARCHING :");
+data=getHashTag(hashTag[i]);
+debug("Data"+data);
+
+  if (data != "") {
+    debug("HASHTAG matched IN DHT");
+             linkHashTags(hashTag[i],data,post); //follow)() changed
+         }
+         else {
+var a=hashTag[i];
+             debug("HASHTAG no match found");
+          //   debug(hashTag[i].value);
+        //  debug(a);
+      //    debug(typeof a);
+             //CREted createHashTag();
+          addHashTag(a);
+           debug("CREATING HASHTAG COMPLETE")
+           debug("ADDING NEW Hanstags:");
+           data=getHashTag(hashTag[i]);
+           debug("Data"+data);
+
+           if (data != "") {
+            debug("HASHTAG FOUND IN DHT");
+                     linkHashTags(hashTag[i],data,post); //follow)() changed
+                 }
+                 else {
+           var a=hashTag;
+                     debug("HASHTAG not found");
+                  //   debug(hashTag[i].value);
+                  debug(a);
+                  debug(typeof a);
+                     //CREted createHashTag();
+                  addHashTag(a);
+                   debug("CREATING Again HASHTAG COMPLETE")
+                 }
+
+       }
+
+
+       getPostsByTag(hashTag[i]); /**THIS METHORD CAN BE USE TO SEARCH THE HASH TAGS
+       <USED here to test if we can get the post to the recently entered hashTag>**/
+
+}
+}
+
+
+function linkHashTags(hashtag,hashTag_Address,post)
+{
+//  getLink(hashTag_Address,"tag_post",{Load:false};
+var hashtagHash = makeHash(hashtag);
+var Entrys = get(hashtagHash,{GetMask:HC.GetMask.Entry});
+var Default=   get(hashtagHash,{GetMask:HC.GetMask.Default});
+var EntryType=get(hashtagHash,{GetMask:HC.GetMask.EntryType});
+var All= get(hashtagHash,{GetMask:HC.GetMask.All});
```

Comment:
It looks like these next lines were added for debugging purposes and now need to be commented out, or deleted..

Just a note FYI, because the GetMask is a bitmask, you coul also call the the get function like this, for example,:  `get(hashtagHash,{GetMask:HC.GetMask.Sources+GetMask.EntryType});` if you just want the Sources and the EntryType